### ISSUE TITLE
thanos compaction failed

### ISSUE THREAD

**Issue by SamPeng87** [thanos-io/thanos#2178]
I don't  know how to resolve this problem.compact component not working now 
```
level=error ts=2020-02-25T03:26:57.268384926Z caller=main.go:194 msg="running command failed" err="error executing compaction: compaction failed: compaction failed for group 0@13248174132169038460: pre compaction overlap check: overlaps found while gathering blocks. [mint: 1582567200000, maxt: 1582574400000, range: 2h0m0s, blocks: 2]: <ulid: 01E1WFZR5A55CKT86BPAYCZHWP, mint: 1582567200000, maxt: 1582574400000, range: 2h0m0s>, <ulid: 01E1WFZR4NVVH4SH2PKR4QRQWR, mint: 1582567200000, maxt: 1582574400000, range: 2h0m0s>\n[mint: 1582581600000, maxt: 1582588800000, range: 2h0m0s, blocks: 2]: <ulid: 01E1WXQ6N72T8QJTE1N3PXDJRC, mint: 1582581600000, maxt: 1582588800000, range: 2h0m0s>, <ulid: 01E1WXQ6MMPSPN3E1WW97EJX22, mint: 1582581600000, maxt: 1582588800000, range: 2h0m0s>\n[mint: 1582545600000, maxt: 1582552800000, range: 2h0m0s, blocks: 2]: <ulid: 01E1VVCJCM5AW16ZN4QB8VCEW0, mint: 1582545600000, maxt: 1582552800000, range: 2h0m0s>, <ulid: 01E1VVCJG0XAGHAR6SBBADA2NE, mint: 1582545600000, maxt: 1582552800000, range: 2h0m0s>\n[mint: 1582552800000, maxt: 1582560000000, range: 2h0m0s, blocks: 2]: <ulid: 01E1W289MMCKPHD5Y4HCF1X0GA, mint: 1582552800000, maxt: 1582560000000, range: 2h0m0s>, <ulid: 01E1W289QY88GFV3WJG09EYGE0, mint: 1582552800000, maxt: 1582560000000, range: 2h0m0s>\n[mint: 1582502400000, maxt: 1582509600000, range: 2h0m0s, blocks: 2]: <ulid: 01E1TJ66WMKRV3Y70CF2WM0XWG, mint: 1582502400000, maxt: 1582509600000, range: 2h0m0s>, <ulid: 01E1TJ66XF40DSP47GDHM8KR2S, mint: 1582502400000, maxt: 1582509600000, range: 2h0m0s>\n[mint: 1582509600000, maxt: 1582516800000, range: 2h0m0s, blocks: 2]: <ulid: 01E1TS1Y4M47907HAC7FC1BWJT, mint: 1582509600000, maxt: 1582516800000, range: 2h0m0s>, <ulid: 01E1TS1Y5KRWBX2ASVD6Y855C9, mint: 1582509600000, maxt: 1582516800000, range: 2h0m0s>\n[mint: 1582516800000, maxt: 1582524000000, range: 2h0m0s, blocks: 2]: <ulid: 01E1TZXNDGHHB29B43EZDVSVYM, mint: 1582516800000, maxt: 1582524000000, range: 2h0m0s>, <ulid: 01E1TZXNDCCGD3ZSBDCGGHCFTA, mint: 1582516800000, maxt: 1582524000000, range: 2h0m0s>\n[mint: 1582524000000, maxt: 1582531200000, range: 2h0m0s, blocks: 2]: <ulid: 01E1V6SCMM8XKWWGS1XENNCPC4, mint: 1582524000000, maxt: 1582531200000, range: 2h0m0s>, <ulid: 01E1V6SCNR93H4J6TME9Y8KRKY, mint: 1582524000000, maxt: 1582531200000, range: 2h0m0s>\n[mint: 1582531200000, maxt: 1582538400000, range: 2h0m0s, blocks: 2]: <ulid: 01E1VDN3XR6PYSMV1TR41FXZMB, mint: 1582531200000, maxt: 1582538400000, range: 2h0m0s>, <ulid: 01E1VDN3WMBTQ1Z561SP1J1H4N, mint: 1582531200000, maxt: 1582538400000, range: 2h0m0s>\n[mint: 1582538400000, maxt: 1582545600000, range: 2h0m0s, blocks: 2]: <ulid: 01E1VMGV4MWE7D94KG7RNZFBR4, mint: 1582538400000, maxt: 1582545600000, range: 2h0m0s>, <ulid: 01E1VMGV86AX7DEGFYRF34132H, mint: 1582538400000, maxt: 1582545600000, range: 2h0m0s>\n[mint: 1582560000000, maxt: 1582567200000, range: 2h0m0s, blocks: 2]: <ulid: 01E1W940WMWJ8215XE08YA600W, mint: 1582560000000, maxt: 1582567200000, range: 2h0m0s>, <ulid: 01E1W940XG0R5JAPGR691YCXKV, mint: 1582560000000, maxt: 1582567200000, range: 2h0m0s>\n[mint: 1582574400000, maxt: 1582581600000, range: 2h0m0s, blocks: 2]: <ulid: 01E1WPVFCM0SPQGBWRPKKRKWDX, mint: 1582574400000, maxt: 1582581600000, range: 2h0m0s>, <ulid: 01E1WPVFD8BQACT338A4EZME02, mint: 1582574400000, maxt: 1582581600000, range: 2h0m0s>
```

**Comment by lrosenman** [comment]
[I] ➜ thanos bucket verify --help
usage: thanos bucket verify [<flags>]

Verify all blocks in the bucket against specified issues

Flags:
  -h, --help               Show context-sensitive help (also try --help-long and --help-man).
      --version            Show application version.
      --log.level=info     Log filtering level.
      --log.format=logfmt  Log format to use.
      --tracing.config-file=<file-path>
                           Path to YAML file with tracing configuration. See format details: https://thanos.io/tracing.md/#configuration
      --tracing.config=<content>
                           Alternative to 'tracing.config-file' flag (lower priority). Content of YAML file with tracing configuration. See format details:
                           https://thanos.io/tracing.md/#configuration
      --objstore.config-file=<file-path>
                           Path to YAML file that contains object store configuration. See format details: https://thanos.io/storage.md/#configuration
      --objstore.config=<content>
                           Alternative to 'objstore.config-file' flag (lower priority). Content of YAML file that contains object store configuration. See format details:
                           https://thanos.io/storage.md/#configuration
      --objstore-backup.config-file=<file-path>
                           Path to YAML file that contains object store-backup configuration. See format details: https://thanos.io/storage.md/#configuration Used for repair logic to backup blocks
                           before removal.
      --objstore-backup.config=<content>
                           Alternative to 'objstore-backup.config-file' flag (lower priority). Content of YAML file that contains object store-backup configuration. See format details:
                           https://thanos.io/storage.md/#configuration Used for repair logic to backup blocks before removal.
  -r, --repair             Attempt to repair blocks for which issues were detected
  -i, --issues=index_issue... ...
                           Issues to verify (and optionally repair). Possible values: [duplicated_compaction index_issue overlapped_blocks]
      --id-whitelist=ID-WHITELIST ...
                           Block IDs to verify (and optionally repair) only. If none is specified, all blocks will be verified. Repeated field


ler in ~ at borg
[I] ➜

**Comment by SamPeng87** [comment]
this command can running any nodes？like storage gateway？

**Comment by SamPeng87** [comment]
level=warn ts=2020-02-25T05:24:05.03844075Z caller=verify.go:49 msg="GLOBAL COMPACTOR SHOULD __NOT__ BE RUNNING ON THE SAME BUCKET" issues=2 repair=false


what mean? compactor only have one ~~

**Comment by SamPeng87** [comment]
msg="repair is not implemented for this issue" issue=overlapped_blocks



how to resolve this?

**Comment by bwplotka** [comment]
Hope this helps: https://github.com/thanos-io/thanos/issues/2182

**Comment by SamPeng87** [comment]
can i auto delete that all overlapped_blocks by command?I can see that uuid is error blocked

**Comment by bwplotka** [comment]
You cannot as only you, as only the owner of this system has knowledge of why overlapped happened and which block to remove to unblock. Also, you should not just remove and forget. Something is wrong with your deployment, you should check out it out. (: 

And at the end, once you know the root cause or want to quickly unblock: Just remove the directory with problematic ULID or move it somewhere else manually via CLI/UI of the object storage of your choice. (: Hope that helps. Closing for now.
